codegen/chains-ts/src/blockchain.ts

```diff
@@ -1,29 +1,173 @@
-export interface Blockchain {
-    readonly name: string;
-    readonly description: string;
-    readonly website: string;
-    readonly explorer: string;
-    readonly logo: string;
+interface BlockchainData {
+    name: string;
+    description: string;
+    website: string;
+    explorer: string;
+    logo: string;
     links: Link[];
     assets: Asset[];
 }
 
-export interface Asset {
-    readonly id: string;
-    readonly name: string;
-    readonly address: string;
-    readonly symbol: string;
-    readonly type: string;
-    readonly assetType: string;
-    readonly bip44CoinType: number;
-    readonly description: string;
-    readonly website: string;
-    readonly explorer: string;
-    readonly decimals: number;
-    readonly icon: string;
+interface AssetData {
+    id: string;
+    name: string;
+    address: string;
+    symbol: string;
+    type: string;
+    assetType: string;
+    bip44CoinType: number;
+    description: string;
+    website: string;
+    explorer: string;
+    decimals: number;
+    icon: string;
 }
 
-export interface Link {
+interface LinkData {
     name: string;
     url: string;
-}
\ No newline at end of file
+}
+
+export class Blockchain {
+    private readonly name: string;
+    private readonly description: string;
+    private readonly website: string;
+    private readonly explorer: string;
+    private readonly logo: string;
+    private readonly links: Link[];
+    private readonly assets: Asset[];
+
+    constructor(data: BlockchainData) {
+        this.name = data.name;
+        this.description = data.description;
+        this.website = data.website;
+        this.explorer = data.explorer;
+        this.logo = data.logo;
+        this.links = data.links;
+        this.assets = data.assets;
+    }
+
+    getName() {
+        return this.name;
+    }
+
+    getDescription() {
+        return this.description;
+    }
+
+    getWebsite() {
+        return this.website;
+    }
+
+    getExplorer() {
+        return this.explorer;
+    }
+
+    getLogo() {
+        return this.logo;
+    }
+
+    getLinks() {
+        return this.links;
+    }
+
+    getAssets() {
```

Comment:
@nima-ca I can't find method `getAsset(id string) {return this.asset}`, please add similar other language.

https://github.com/ezex-io/chains/blob/714fa78e7a43f1c8a3c34e368b7782482416e612/codegen/chains-rs/src/pactus.rs#L54-L59